include/libudffs.h

```diff
@@ -229,7 +229,7 @@ extern size_t decode_string(struct udf_disc *, const dstring *, char *, size_t,
 extern size_t encode_string(struct udf_disc *, dstring *, const char *, size_t);
 
 /* misc.c */
-extern const char *appname;
+const char *appname;
```

Comment:
This is wrong change as variable appname is already defined in misc.c. Therefore it can be only declared, not defined.

Reply:
This is the kind of error I get in folder mkudffs, if I don't drop that extern:

/bin/sh ../libtool  --tag=CC   --mode=link gcc  -g -O2   -o mkudffs main.o mkudffs.o defaults.o file.o options.o ../libudffs/libudffs.la
libtool: link: gcc -g -O2 -o mkudffs main.o mkudffs.o defaults.o file.o options.o  ../libudffs/.libs/libudffs.a
Undefined symbols for architecture x86_64:
  "_appname", referenced from:
      _main in main.o
      _split_space in mkudffs.o
      _setup_anchor in mkudffs.o
      _setup_partition in mkudffs.o
      _setup_space in mkudffs.o
      _setup_vat in mkudffs.o
      _setup_vds in mkudffs.o

Reply:
It is defined in misc.c which is linked into libudffs.a. If linker does not see appname in libudffs.a when building mkudffs binary, you would need to debug build process and find out what is happening there.

I do not know Apple nor Darwin systems, so I cannot help with this part.